### ISSUE TITLE
MetaDEx Fees Payment Property

### ISSUE THREAD

**Issue by zathras-crypto** [OmniLayer/omnicore#350]
Please consider the following:
* Alice sells 2000 [5] for 2000 [6]
* Bob sells 2000 [6] for 2000 [5]
* The trade is matched and executed.
* Bob is removing liquidity from the market.  
* He needs to pay a fee in the "payment" property of 1 [6] =(0.05%). 

What is the intended handling?

If we take 1 [6] for fee after the trade matches, there is only 1999 [6] left to give to Alice.  Alice ends up paying the fee when Bob is the liquidity taker.

However if we instead take 1 [5] from Bob after the trade matches, Alice receives 2000 [6] as intended and Bob receives 1999 [5].

The difference here is that @patrickdugan requested the fee be paid in the liquidity takers payment property (in Omni terms the later offers propertyforsale).  I don't believe this is feasible to do without  throwing off the math in the trade (ie 2000/2000 becomes 1999/2000) or transferring the fee reduction to the other party - at least with the direction I'm currently taking for fees anyway (@dexx7 if you want to correct me here please shout)

If we simply say the fee is paid in the property the liquidity taker *receives* instead, it resolves the entire issue.  These tokens are inbound to the liquidity taker and thus it's a simple case to take 0.05% of them as fee and giving 1 less.  If we do it the other way around and take 0.05% of Bob's outbound tokens, there is no longer enough to pay Alice fully.

Hope that makes sense ! :)

TL:DR; I'd like to propose that the fee is collected in the property the liquidity taker (the later offer) receives, instead of the "payment" property.

**Comment by patrickdugan** [comment]
Nice catch, I imagined this but wrote it down differently later.

In a word: yes.

**Comment by zathras-crypto** [comment]
Perfect, thanks.

Closing...
